### PR TITLE
feat: leveraging angular built-in tailwind support & setting purge.en…

### PR DESCRIPTION
**Pull request by vltansky**
…abled

### CODE REVIEW

File: apps/tailwind-e2e/tests/tailwind.spec.ts [hunk 1]
```diff
@@ -26,10 +26,6 @@ async function assertNx(schematicName: 'nx-setup' | 'ng-add') {
 
   await runNxCommandAsync(`generate @ngneat/tailwind:${schematicName}`);
 
-  expect(() =>
-    checkFilesExist('tailwind.config.js', 'webpack.config.js')
-  ).not.toThrow();
-
```
Comment (nartc) [inline apps/tailwind-e2e/tests/tailwind.spec.ts:32]: question: should we still need to check for `tailwind.config.js`?

File: libs/tailwind/src/schematics/files/tailwind.config.js.template [hunk 2]
```diff
@@ -1,5 +1,8 @@
+const { guessProductionMode } = require("@ngneat/tailwind");
+
 module.exports = {
     prefix: '',
+    enabled: guessProductionMode(),
```
Comment (nartc) [inline libs/tailwind/src/schematics/files/tailwind.config.js.template:5]: issue: `enabled` is under `purge`, not on the same level as `purge`

**Review by nartc — CHANGES_REQUESTED** [review]

File: apps/tailwind-e2e/tests/tailwind.spec.ts (hunk 1 above)
Comment (vltansky) [reply apps/tailwind-e2e/tests/tailwind.spec.ts:32]: We do. Removed it by mistake :)

**Comment by nartc** [comment]
LGTM

**Comment by vltansky** [comment]
So this PR resolves the following tasks of roadmap:

- [ ] ngneat/tailwind will check if your project is running v11.2 of Angular.
If yes, then ngneat/tailwind will execute a slightly newer flow
If no, then ngneat/tailwind will execute the current flow, which is working for most Angular projects that are under 11.2

- [x] ngneat/tailwind will ride on top of AngularCLI support for Tailwind. This means that we will not leverage any custom webpack anymore.
- [x] ngneat/tailwind will still initialize a tailwind.config.js for you without you having to execute npx tailwindcss init
- [x] ngneat/tailwind will also setup Purging out of the box with a custom flag.
- [ ] ngneat/tailwind will export a custom Tailwind plugin to handle ViewEncapsulation in Components' styles. Right now, this is handled by a custom PostCSS plugin but since we won't be using Custom Webpack, we cannot utilize a PostCSS plugin.
- [x] ngneat/tailwind will export the Custom Tailwind Plugin and the Purging flag as Public API so the consumers can reference if need be.

I think this can be merged into master, as the tailwind plugin & angular version checker can be set up as separate PRs

#79

**Comment by nartc** [comment]
Thanks @vltansky

**Comment by twittwer** [comment]
Is there an estimate when this PR will be released? 🙂

**Pull request merged**